### ISSUE TITLE
[Question] Stability of embeddings on consecutive runs?

### ISSUE THREAD

**Issue by ml7** [facebookresearch/PyTorch-BigGraph#264]
Hi there! I hope all is well. I noticed in the code that the embeddings seem to be initialized with torch.empty, which appears to give slightly different results on each call (both in terms of magnitude and orientation). We're noticing that the resulting embeddings trained on two separate runs (holding data fixed) seem to differ as a result. We're wondering if the initialization might be the cause.

Or would it also potentially be caused by the negative sampling not producing the same negatives? Any thoughts are appreciated!

**Comment by parsa-saadatpanah** [comment]
There are a few random operations (including parameter initialization and negative sampling) involved in the training process that would contribute to different trained embeddings.
Unfortunately setting the random seed is currently not supported.

**Comment by ml7** [comment]
Thank you @parsa-saadatpanah! Could you elaborate on some of the other random operations you see that could contribute to different trained embeddings? It would help me figure out what we can do on our end to adjust. Thanks!

**Comment by parsa-saadatpanah** [comment]
I believe the bucket scheduling and how each bucket is split into sub-buckets can also be random.
